includes/ucf-section-common.php

```diff
@@ -173,7 +173,100 @@ public static function format_shortcode_output( $content ) {
 			$rep = preg_replace( "/(<p>)?\[\/($block)](<\/p>|<br \/>)?/", "[/$2]", $rep );
 			return $rep;
 		}
+
+		/**
+		 * Called by the wp_head filter
+		 * Determines if a section has a stylesheet to print
+		 * into the head of the document
+		 * @author Jim Barnes
+		 * @since 1.0.1
+		 **/
+		public static function get_section_stylesheets() {
+			global $post;
+
+			wp_enqueue_style( 'section-styles', plugins_url( 'static/css/empty.css', UCF_SECTION__PLUGIN_FILE ) );
+			wp_enqueue_script( 'section-scripts', plugins_url( 'static/js/empty.js', UCF_SECTION__PLUGIN_FILE ), array( 'jquery' ), null, TRUE );
+
+			$styles_to_print = array();
+			$scripts_to_print = array();
+
+			if ( has_shortcode( $post->post_content, 'ucf-section' ) ) {
+				$pattern = get_shortcode_regex( array( 'ucf-section' ) );
+
+				preg_match_all( '/' . $pattern . '/s', $post->post_content, $matches );
+
+				if ( preg_match_all( '/' . $pattern . '/s', $post->post_content, $matches ) &&
+					array_key_exists( 3, $matches ) ) {
+
+					foreach( $matches[3] as $match ) {
+						$args = shortcode_parse_atts( $match );
+
+						$section = null;
+
+						if ( isset( $args['slug'] ) ) {
+							$section = self::get_section_by_slug( $args['slug'] );
+						}
+
+						if ( isset( $args['id'] ) ) {
+							$section = get_post( $args['id'] );
+						}
+
+						if ( $section !== null ) {
+							$stylesheet_id = get_post_meta( $section->ID, 'ucf_section_stylesheet', TRUE );
+
+							if ( $stylesheet_id && ! key_exists( $styles_to_print, $stylesheet_id ) ) {
+								$styles_to_print[$stylesheet_id] = $stylesheet_id;
+							}
+
+							$javascript_id = get_post_meta( $section->ID, 'ucf_section_javascript', TRUE );
+
+							if ( $javascript_id && ! key_exists( $scripts_to_print, $javascript_id ) ) {
+								$scripts_to_print[$javascript_id] = $javascript_id;
+							}
+						}
+					}
+				}
+
+				// Go ahead and print each stylesheet
+				foreach( $styles_to_print as $id ) {
+					self::print_stylesheet_to_head( $id );
+				}
+
+				// Go ahead and print each javascript
+				foreach( $scripts_to_print as $id ) {
+					self::print_javascript_to_footer( $id );
+				}
+			}
+		}
+
+		/**
+		 * Retrieves the stylesheet and prints it to head
+		 * @author Jim Barnes
+		 * @since 1.0.4
+		 * @param int $stylesheet_id | The ID of the stylesheet to print
+		 **/
+		private static function print_stylesheet_to_head( $stylesheet_id ) {
+			$style_filepath = get_attached_file( $stylesheet_id );
+			$contents = file_get_contents( $style_filepath );
```

Comment:
Can `wp_remote_get()` be used here (and line 263) instead to avoid usage of `file_get_contents()`?

Reply:
Also, a check to ensure $style_filepath (and $script_filepath) are set properly would be helpful to avoid passing potentially invalid URLs/paths to `file_get_contents()`

Reply:
In this case, because we're accessing only local files, and the file paths are provided directly by WordPress, using `file_get_contents` is okay. The url_fopen security setting only prevents file_get_contents from pulling remote files. We might want to be careful about using this if the path was being determined based on some sort user input.